src/CMakeLists.txt

```diff
@@ -223,9 +229,34 @@ endif()
 
 target_link_libraries(parthenon PUBLIC Kokkos::kokkos)
 
+lint_target(parthenon)
+
 target_include_directories(parthenon PUBLIC
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_CURRENT_BINARY_DIR}/generated
-)
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/generated>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  )
+
+install(TARGETS parthenon EXPORT parthenonTargets
+  INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/parthenon"
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
+
+# Maintain directory structure in installed include files
+install(DIRECTORY . DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.hpp")
+
+# Install generated config header file
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/generated/config.hpp
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+configure_file("${PROJECT_SOURCE_DIR}/cmake/parthenonConfig.cmake.in" "${PROJECT_BINARY_DIR}/cmake/parthenonConfig.cmake" @ONLY)
```

Comment:
This `configure_file` bit should logically go in the root `CMakeLists.txt`, I think, since the config is "project" wide rather than just for this specific library.